models/articles-model.js

```diff
@@ -95,19 +94,34 @@ exports.updateArticleByID = (article_id, inc_votes) => {
 exports.fetchCommentsByID = (article_id) => {
   const existingArticles = [];
 
-  db.query(`SELECT article_id FROM articles`).then(({rows: articles}) => articles.forEach(article => existingArticles.push(article)));
-  
+  db.query(`SELECT article_id FROM articles`).then(({ rows: articles }) =>
+    articles.forEach((article) => existingArticles.push(article))
+  );
+
+  return db
+    .query(
+      `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;`,
+      [article_id]
+    )
+    .then(({ rows: comments }) => {
+      if (article_id > existingArticles.length) {
+        return Promise.reject({ status: 404, msg: "Article not found" });
+      }
+      if (comments.length === 0 && article_id <= existingArticles.length) {
+        return Promise.reject({
+          status: 200,
+          msg: "No comments found for that article yet",
```

Comment:
it's ok to return the empty array from SQL for 0 comments of an existing article

Reply:
Will do, thank you Alex! 😁👌